### PR TITLE
feat(wallet-ui): add data fetching and refactor routes

### PR DESCRIPTION
**Pull request by gregfromstl**
This PR adds new UI components and various utilities for fetching blockchain data, leveraging third-party libraries such as thirdweb and simplehash. It also introduces new dependencies to the project.

### What changed?

- Add `@tanstack/react-query` and `pino-pretty` dependencies.
- Create new layout components for different sections within the `wallet-ui` app.
- Introduce blockchain data fetching utilities for ERC20, ERC721, and native token balances.
- Update and reorganize routing logic to reflect new UI and authentication flows.

### How to test?

1. Install the new dependencies by running `pnpm install`.
2. Navigate through the updated wallet UI to ensure the new layout and routing logic work as expected.
3. Verify the blockchain data fetching utilities by checking console logs or integrating them further into your components.

### Why make this change?

The motivation behind these changes is to improve the wallet UI by adding new layouts and making it capable of fetching and displaying various types of blockchain data efficiently. These updates aim to enhance user experience and provide more functionality.

---

---

## PR-Codex overview
The focus of this PR is to add Thirdweb integration, update dependencies, improve UI components, and enhance ecosystem handling.

### Detailed summary
- Added Thirdweb integration for wallet functionality
- Updated dependencies for better performance
- Improved UI components like Skeleton, Card, and Toast
- Enhanced ecosystem handling with ecosystem-specific pages
- Included Tailwind configuration for styling
- Added providers for global state management

> The following files were skipped due to too many changes: `apps/wallet-ui/src/lib/chains.ts`, `apps/wallet-ui/src/components/ui/input.tsx`, `apps/wallet-ui/src/components/ui/badge.tsx`, `apps/wallet-ui/src/app/[ecosystem]/(ui)/layout.tsx`, `apps/wallet-ui/src/components/ui/popover.tsx`, `apps/wallet-ui/src/app/[ecosystem]/(ui)/wallet/[address]/layout.tsx`, `apps/wallet-ui/src/lib/assets/erc20.ts`, `apps/wallet-ui/src/components/ui/button.tsx`, `apps/wallet-ui/src/components/NftGallery.tsx`, `apps/wallet-ui/src/components/NftCard.tsx`, `apps/wallet-ui/src/lib/assets/erc721.ts`, `apps/wallet-ui/src/util/simplehash.ts`, `apps/wallet-ui/src/components/ChainCombobox.tsx`, `apps/wallet-ui/src/components/TransactionToastProvider.tsx`, `apps/wallet-ui/src/components/ui/use-toast.ts`, `apps/wallet-ui/src/app/globals.css`, `apps/wallet-ui/src/components/ui/dialog.tsx`, `apps/wallet-ui/src/components/ChainIcon.tsx`, `apps/wallet-ui/src/components/ui/form.tsx`, `apps/wallet-ui/src/components/NftModal.tsx`, `apps/wallet-ui/src/components/ui/toast.tsx`, `apps/wallet-ui/src/components/ui/command.tsx`, `pnpm-lock.yaml`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: apps/wallet-ui/package.json [hunk 1]
```diff
@@ -10,7 +10,9 @@
     "fix": "biome check ./src --fix"
   },
   "dependencies": {
+    "@tanstack/react-query": "^5.48.0",
     "next": "14.2.4",
+    "pino-pretty": "^11.2.1",
```
Comment (jnsdls) [inline apps/wallet-ui/package.json:15]: definitely not a dependency, if this is shouting at you in logs check playground how we disabled it in `next config`

File: apps/wallet-ui/src/app/[ecosystem]/(ui)/layout.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import { client } from "@/lib/client";
+import { useRouter } from "next/navigation";
+import { ConnectButton, useActiveAccount } from "thirdweb/react";
+
+export default function Layout({
+  children,
+  params,
+}: { children: React.ReactNode; params: { ecosystem: string } }) {
+  return (
+    <div className="">
```
Comment (jnsdls) [inline apps/wallet-ui/src/app/[ecosystem]/(ui)/layout.tsx:11]: nit: empty className is weird

File: apps/wallet-ui/src/app/[ecosystem]/(ui)/wallet/[address]/page.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { getErc20Tokens } from "@/lib/assets/erc20";
+import { getErc721Tokens } from "@/lib/assets/erc721";
+import { SIMPLEHASH_SUPPORTED_CHAIN_IDS } from "@/util/simplehash";
+import { isAddress } from "thirdweb";
+import { getNativeBalances } from "../../../../../lib/assets/native";
+
+export default async function Page({
+  params: { address },
```
Comment (jnsdls) [inline apps/wallet-ui/src/app/[ecosystem]/(ui)/wallet/[address]/page.tsx:8]: hm, driving this from URL is a bit odd, it means that anyone can look at anyone's wallet contents via this page -- technically maybe fine but not what I would've expected

Comment (jnsdls) [reply apps/wallet-ui/src/app/[ecosystem]/(ui)/wallet/[address]/page.tsx:8]: my expecation here would be:

1. I connect wallet
2. I sign in (backend now has a cookie to work with)
3. the page here is driven by the address on the cookie

File: apps/wallet-ui/src/app/[ecosystem]/login/layout.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import { useRouter } from "next/navigation";
+import { useActiveAccount } from "thirdweb/react";
+
+export default function Layout({ children }: { children: React.ReactNode }) {
+  const router = useRouter();
+  const account = useActiveAccount();
+
+  if (account) {
+    router.push(`/wallet/${account.address}`); // This will route to /[ecosystem]/wallet/[address] based on the subdomain routing
+  }
```
Comment (jnsdls) [inline apps/wallet-ui/src/app/[ecosystem]/login/layout.tsx:11]: again this is problematic RE access control

File: apps/wallet-ui/src/app/[ecosystem]/login/page.tsx [hunk 5]
```diff
@@ -0,0 +1,14 @@
+"use client";
+import { client } from "@/lib/client";
+import { ConnectEmbed, useActiveAccount } from "thirdweb/react";
+import { inAppWallet } from "thirdweb/wallets";
+
+export default function Page() {
+  return (
+    <ConnectEmbed
+      autoConnect={true}
+      wallets={[inAppWallet()]}
```
Comment (jnsdls) [inline apps/wallet-ui/src/app/[ecosystem]/login/page.tsx:10]: would have to be ecosystem wallet right? with the correct ID etc?

File: apps/wallet-ui/src/app/[ecosystem]/page.tsx [hunk 6]
```diff
@@ -1,3 +1,17 @@
...
+  const account = useActiveAccount();
+  const router = useRouter();
+
+  if (!account) {
+    router.push("/login"); // This will route to /[ecosystem]/login based on the subdomain routing
+  } else {
+    router.push(`/${account.address}/`);
+  }
+
+  return null;
```
Comment (jnsdls) [inline apps/wallet-ui/src/app/[ecosystem]/page.tsx:16]: yeah this should be happening on server side

File: apps/wallet-ui/src/components/Providers.tsx [hunk 7]
```diff
@@ -0,0 +1,13 @@
+"use client";
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import { ThirdwebProvider } from "thirdweb/react";
+
+const queryClient = new QueryClient();
+
+export default function Providers({ children }: { children: React.ReactNode }) {
+  return (
+    <ThirdwebProvider>
+      <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
```
Comment (jnsdls) [inline apps/wallet-ui/src/components/Providers.tsx:10]: out of interest, what is the separate queryClient provider for?

**Review by jnsdls — CHANGES_REQUESTED** [review]

File: apps/wallet-ui/src/app/[ecosystem]/login/page.tsx (hunk 5 above)
Comment (gregfromstl) [reply apps/wallet-ui/src/app/[ecosystem]/login/page.tsx:10]: Yep this is temporary

**Pull request merged**